controllers/k8ssandracluster_controller.go

```diff
@@ -412,20 +416,30 @@ func (r *K8ssandraClusterReconciler) ensureStargateAuthKeyspaceExists(
 ) error {
 	// TODO create a endpoint in the Management API to list existing keyspaces and test whether the keyspace
 	// already exists.
-	logger.Info(fmt.Sprintf("Ensuring that keyspace data_endpoint_auth exists in cluster %v...", dc.ClusterName))
-	replication := make(map[string]int, len(kc.Spec.Cassandra.Datacenters))
-	for _, dcTemplate := range kc.Spec.Cassandra.Datacenters {
-		replicationFactor := int(math.Min(3.0, float64(dcTemplate.Size)))
-		replication[dcTemplate.Meta.Name] = replicationFactor
-	}
 	if managementApi, err := r.ManagementApi.NewManagementApiFacade(ctx, dc, remoteClient, logger); err != nil {
 		return err
-	} else if err := managementApi.CreateKeyspaceIfNotExists("data_endpoint_auth", replication); err != nil {
-		logger.Error(err, "Failed to create keyspace data_endpoint_auth")
-		return err
 	} else {
-		logger.Info("Keyspace data_endpoint_auth successfully created, or already exists")
-		return nil
+		logger.Info(fmt.Sprintf("Ensuring that keyspace %s exists in cluster %v...", STARGATE_AUTH_KEYSPACE, dc.ClusterName))
+		if keyspaces, err := managementApi.ListKeyspaces(STARGATE_AUTH_KEYSPACE); err != nil {
+			return err
+		} else if len(keyspaces) == 0 {
+			// Stargate auth keyspace doesn't exist, we need to create it.
+			replication := make(map[string]int, len(kc.Spec.Cassandra.Datacenters))
+			for _, dcTemplate := range kc.Spec.Cassandra.Datacenters {
+				replicationFactor := int(math.Min(3.0, float64(dcTemplate.Size)))
+				replication[dcTemplate.Meta.Name] = replicationFactor
+			}
+			if err := managementApi.CreateKeyspaceIfNotExists("data_endpoint_auth", replication); err != nil {
```

Comment:
This line uses a magical string instead of the constant that was defined.

Reply:
I've lost some changes during a rebase it seems, good catch 👍

Reply:
fixed